### PR TITLE
Uncertain dates with negative years do not parse.

### PR DESCRIPTION
**Pull request by workergnome**
Given a EDTF string like "-1990-uu-uu", the library currently does not parse it.  I've added a failing test and a modification to the parser to accommodate dates like this.

I'm not 100% sure that this is valid according to the EDTF spec, since the BNF appears to specify 'year' in two different ways:

```
(* ** Definition for year ** *)

year = positiveYear | negativeYear | "0000"

positiveYear = 
       positiveDigit digit digit digit 
     | digit positiveDigit digit digit
     | digit digit positiveDigit digit
     | digit digit digit positiveDigit

negativeYear = "-" positiveYear


(* ** Other Auxiliary Assignments for Level 0 ** *)
year = digit digit digit digit
```

That being said, the written text appears to state that this would and should be valid.

### CODE REVIEW

**Comment by workergnome** [comment]
So.  It doesn't appear to work with -19uu-uu-uu, and I can't for the life of me figure out why.  Played around with the parser for a while, but I couldn't figure it out.

There's a failing test, and if you have a suggestion, I'm happy to keep playing...

File: lib/edtf/parser.y [hunk 1]
```diff
@@ -131,6 +131,8 @@ rule
     result = Date.new(val[0]).unspecified!([:day,:month])
   }
 
+  negative_unspecified :  '-' unspecified { result = -val[1] }
```
Comment (inukshuk) [inline lib/edtf/parser.y]: Here `val[1]` is a an array of values, so here you actually invert only `val[1][0]`.

**Comment by inukshuk** [comment]
Thanks! I added the parsing support for negative years in `2348952` -- can you try there?

Whenever you're puzzled by the parser, it's best to re-compile it using `bundle exec rake racc_debug` and initialize / set the parser's debug option before parsing -- that will print all the intermediary parser states.

**Comment by workergnome** [comment]
Thanks—`2348952` looks like it fixes my issue.  And thanks for the hint about the parser—I'll give that a shot next time.

**Comment by inukshuk** [comment]
Cool, thanks, then I'll push a new release to RubyGems.

**Pull request closed without merging**